### PR TITLE
Encoder common separation, PR 6

### PR DESCRIPTION
**Pull request by andrey-norkin**
This PR continues separation of the encoder library into the encoder and common libraries. This PR mostly includes separation of the superresolution tool and Intra prediction.

### CODE REVIEW

File: Source/Lib/Common/Codec/EbIntraPrediction.h [hunk 1]
```diff
@@ -6,8 +6,12 @@
...
 #define EbIntraPrediction_h
 
-#include "EbModeDecision.h"
-#include "EbMotionEstimationProcess.h"
+#include "EbSvtAv1.h"
+#include "EbObject.h"
+#include "EbBlockStructures.h"
+
+//#include "EbModeDecision.h"
+//#include "EbMotionEstimationProcess.h"
```
Comment (hassount) [inline Source/Lib/Common/Codec/EbIntraPrediction.h]: can you please remove the extra spaces and commented includes

File: Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.h [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include "EbNeighborArrays.h"
 #include "EbMvMerge.h"
 #include "EbWarpedMotion.h"
+//#include "EbModeDecisionProcess.h"
```
Comment (hassount) [inline Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.h]: remove

File: Source/Lib/Common/Codec/EbIntraPrediction.h (hunk 1 above)
Comment (andrey-norkin) [reply Source/Lib/Common/Codec/EbIntraPrediction.h]: Done

File: Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.h (hunk 2 above)
Comment (andrey-norkin) [reply Source/Lib/Encoder/Codec/EbAdaptiveMotionVectorPrediction.h]: Done

**Comment by hassount** [comment]
Checked bitstream exactness on -enc-mode 0,4,8

**Pull request merged**